internetarchive/config.py

```diff
@@ -70,7 +71,7 @@ def get_auth_config(email, password, host=None):
     return auth_config
 
 
-def write_config_file(auth_config, config_file=None):
+def write_config_file(auth_config: Mapping, config_file=None):
```

Comment:
You might want to use `mapping` instead of `Mapping`, depending on how you wrote it in the other PRs, but if I recall correctly you decided to use the lowercase version of all the types.

`-> string`, but should probably be some proper path type in the future, so maybe keep it the way it is for now.